specification.md

```diff
@@ -1631,6 +1631,15 @@ There are two types of conditions for switch states:
 
 These are exclusive, meaning that a switch state can define one or the other condition type, but not both.
 
+The order on which conditions are defined must be respected by implementations unless priorities are speficied. 
+If priorities are specified, the most prioritary among satisfied conditions must have its transition executed.
+
+For example, let's say there are two conditions, A and B, defined in that order. If both A and B are satisfied
+then the A's transition must be executed. This behavior is akin to swtich statements in many programming languages.
+On the other hand, if A had a priority of 10 and B's priority was 20 then B's transition would be executed instead.
+
+If the `priority` parameter of one of the defined conditions is specified, all conditions must also specify it.
```

Comment:
Not sure if this is true. All other nodes could have `integer.MAX` defined by default. Would improve UX, I don't want to start adding a new field in all my states just because I need two or more states to have prio.

Reply:
This is per state though, all conditions in the same state must specify priority. I can clarify that, thanks for pointing it out!

But do you think this is still undesirable within a single Switch State?

Int.MAX would effectively make all undefined priorities to take precedence over those with a priority. Wouldn't int.min be more reasonable?

Come to think of it, this parameter could probably use some boundaries. [0..100] for example, wdyt?

Reply:
I don't think we need to go as far as defining boundaries. The only restriction I would place is "positive integer including 0", so I would definitely add "minimum": 0 to the definition in json schema.

Reply:
To me, priority 0 takes over 1. So, if all other states or actors in this context have a greater number they will have lower priority.

Like @tsurdilo said minimum 0 in the schema will do the job and every other state that does not have this attribute, it's assumed it will take the lower priority.

Reply:
I think bpmn works like that (lower priority meaning higher rank) but we don't have to follow that. I think from just the language point of view or way we think, higher priority means higher rank. So we can just go with that for our lang def. What would be the counter-argument to this?

Reply:
To me lower numbers for higher priority seems a bit counter-intuitive, but historically there are plenty of examples choosing either way, so I don't think there would be a problem with any choice.

@tsurdilo any thoughts about this?

Reply:
yeah like i said to me its more logical that higher priority value means higher rank (gets picked first).